### PR TITLE
adding cause tag for transient poll failures.

### PR DESCRIPTION
**Pull request by mindaugasbarcauskas**
Adding tags to be able to discern if poll failures are due to rate limiting or service failures.

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java [hunk 1]
```diff
@@ -66,10 +70,18 @@ protected PollForActivityTaskResponse pollTask() throws TException {
     PollForActivityTaskResponse result;
     try {
       result = service.PollForActivityTask(pollRequest);
-    } catch (InternalServiceError | ServiceBusyError e) {
-      options.getMetricsScope().counter(MetricsType.ACTIVITY_POLL_TRANSIENT_FAILED_COUNTER).inc(1);
+    } catch (InternalServiceError e) {
+      Map<String, String> tags = new HashMap<>();
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java:74]: nit: please consider ImmutableMap.Builder<String, String> instead

File: src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java [hunk 2]
```diff
@@ -66,10 +70,18 @@ protected PollForActivityTaskResponse pollTask() throws TException {
...
+      tags.put(MetricsTag.CAUSE, "internalServiceError");
+      options.getMetricsScope().tagged(tags).counter(MetricsType.ACTIVITY_POLL_TRANSIENT_FAILED_COUNTER).inc(1);
+      throw e;
+    } catch (ServiceBusyError e) {
+      Map<String, String> tags = new HashMap<>();
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java:79]: idem

File: src/main/java/com/uber/cadence/internal/worker/WorkflowPollTask.java [hunk 3]
```diff
@@ -77,8 +80,15 @@ public PollForDecisionTaskResponse poll() throws TException {
     PollForDecisionTaskResponse result;
     try {
       result = service.PollForDecisionTask(pollRequest);
-    } catch (InternalServiceError | ServiceBusyError e) {
-      metricScope.counter(MetricsType.DECISION_POLL_TRANSIENT_FAILED_COUNTER).inc(1);
+    } catch (InternalServiceError e) {
+      Map<String, String> tags = new HashMap<>();
+      tags.put(MetricsTag.CAUSE, "internalServiceError");
+      metricScope.tagged(tags).counter(MetricsType.DECISION_POLL_TRANSIENT_FAILED_COUNTER).inc(1);
+      throw e;
```
Comment (mkolodezny) [inline src/main/java/com/uber/cadence/internal/worker/WorkflowPollTask.java:87]: nit: could be moved to a separate method to dedup the above

**Review by mkolodezny — APPROVED** [review]

File: src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java (hunk 1 above)
Comment (mindaugasbarcauskas) [reply src/main/java/com/uber/cadence/internal/worker/ActivityPollTask.java]: done.

**Pull request merged**